staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit_annotations.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"net/http"
+
+	"k8s.io/apiserver/pkg/audit"
+	"k8s.io/apiserver/pkg/audit/policy"
+)
+
+// WithAuditAnnotations decorates a http.Handler with a []{key, value} that is merged
+// with the audit.Event.Annotations map.  This allows layers that run before WithAudit
+// (such as authentication) to assert annotations.
+// If sink or audit policy is nil, no decoration takes place.
+func WithAuditAnnotations(handler http.Handler, sink audit.Sink, policy policy.Checker) http.Handler {
```

Comment:
It's pretty strange that these are passed in but not used. Can we change this to a `enabled bool` or just not call this function if audit is disabled?

Reply:
The bulk of our handlers check their inputs and do no wrapping when disabled (ex: `WithAudit`, `WithAuthentication`):

https://github.com/kubernetes/kubernetes/blob/0f76d5c5eb6d45acf20027a272b4f8ed418889bc/staging/src/k8s.io/apiserver/pkg/server/config.go#L660-L682

To me it would be better to get rid of those two `if` statements so that chain is easier to read.  Most of our wrappers are already `MaybeWithFoo` style functions.  I do not want to diverge from that.